test/conformance/api/v1/service_test.go

```diff
@@ -581,3 +582,56 @@ func TestAnnotationPropagation(t *testing.T) {
 		t.Error("Route still has `juicy` annotation")
 	}
 }
+
+// TestServiceCreateWithMultipleContainers tests both Creation paths for a service.
+// The test performs a series of Validate steps to ensure that the service transitions as expected during each step.
+func TestServiceCreateWithMultipleContainers(t *testing.T) {
+	if !test.ServingFlags.EnableAlphaFeatures {
+		return
+	}
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+	}
+
+	// Clean up on test failure or interrupt
+	test.EnsureTearDown(t, clients, &names)
+	// images are used to validate digest in validateControlPlane function
+	images := map[string]string{
+		"serving-container": test.ServingContainer,
+		"sidecar-container": test.SidecarContainer,
+	}
+	containers := []corev1.Container{{
+		Name:  "serving-container",
+		Image: pkgTest.ImagePath(test.ServingContainer),
+		Ports: []corev1.ContainerPort{{
+			ContainerPort: 8881,
+		}},
+	}, {
+		Name:  "sidecar-container",
+		Image: pkgTest.ImagePath(test.SidecarContainer),
+	}}
+
+	// Setup initial Service
+	objects, err := v1test.CreateServiceReadyForMultiContainer(t, clients, &names, func(svc *v1.Service) {
+		svc.Spec.Template.Spec.Containers = containers
+	})
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	// Validate State after Creation
+	if err = validateControlPlane(t, clients, names, "1" /*1 is the expected generation value*/, images); err != nil {
+		t.Error(err)
+	}
+
+	if err = validateDataPlane(t, clients, names, test.MultiContainerResponse); err != nil {
+		t.Error(err)
+	}
+
+	if err = validateLabelsPropagation(t, *objects, names); err != nil {
```

Comment:
Is this anything special for multi-containers? If not, let's ditch it.

Reply:
+1